### PR TITLE
fix(install): improve powershell script (Issue #372)

### PR DESCRIPTION
**Pull request by KrishPatel13**
**What kind of change does this PR introduce?**
This PR resolves the bugs numbered here https://github.com/OpenAdaptAI/OpenAdapt/issues/372#issue-1796969872

**Summary**
Bug Number fixed:

- [x] [1](https://github.com/OpenAdaptAI/OpenAdapt/pull/409#issuecomment-1642692405)
- [x] [2](https://github.com/OpenAdaptAI/OpenAdapt/pull/409#issuecomment-1642692405)
- [x] [3](https://github.com/OpenAdaptAI/OpenAdapt/pull/409#discussion_r1268653524) (further information is needed for resolution) So to mark it as resolved for now read: https://github.com/OpenAdaptAI/OpenAdapt/pull/409#discussion_r1268653524
- [x] [4](https://github.com/OpenAdaptAI/OpenAdapt/pull/409#issuecomment-1642689392)
- [x] [5](https://github.com/OpenAdaptAI/OpenAdapt/pull/409#issuecomment-1642697181)
- [x] [6](https://github.com/OpenAdaptAI/OpenAdapt/pull/409#issuecomment-1642697181)

**Checklist**
* [x] My code follows the style guidelines of OpenAdapt
* [x] I have performed a self-review of my code
* [x] If applicable, I have added tests to prove my fix is functional/effective
* [x] I have linted my code locally prior to submission
* [x] I have commented my code, particularly in hard-to-understand areas
* [x] I have made corresponding changes to the documentation (e.g. README.md, requirements.txt)
* [x] New and existing unit tests pass locally with my changes

**How can your code be run and tested?**

**Other information**

### CODE REVIEW

**Comment by KrishPatel13** [comment]
Bug 4 (from https://github.com/OpenAdaptAI/OpenAdapt/issues/372#issue-1796969872)
image

Resolution Transcript: 
```
PS C:\Users\Krish Patel> python -V
Python 3.10.11
PS C:\Users\Krish Patel> $pythonMinVersion = "3.10.0" # Change this if a different Lower version are supported by OpenAdapt
PS C:\Users\Krish Patel> $pythonMaxVersion = "3.10.12" # Change this if a different HIgher version are supported by OpenAdapt
PS C:\Users\Krish Patel> function ComparePythonVersion($version) {
>>     $v = [version]::new($version)
>>     $min = [version]::new($pythonMinVersion)
>>     $max = [version]::new($pythonMaxVersion)
>>
>>     return $v -ge $min -and $v -le $max
>> }
PS C:\Users\Krish Patel> $res = Invoke-Expression "python -V"
PS C:\Users\Krish Patel>         $versionString = $res.Split(' ')[-1]
PS C:\Users\Krish Patel> $res
Python 3.10.11
PS C:\Users\Krish Patel> $versionString
3.10.11
PS C:\Users\Krish Patel> if (ComparePythonVersion $versionString  $pythonMaxVersion) {
>>             return $true
>>         }
True
PS C:\Users\Krish Patel> if (ComparePythonVersion $versionString  $pythonMaxVersion) {
>>             return "correct version found"
>>         }
correct version found
PS C:\Users\Krish Patel> $versionString = "3.11.0"
PS C:\Users\Krish Patel> if (ComparePythonVersion $versionString  $pythonMaxVersion) {
>>             return "correct version found"
>>         }
PS C:\Users\Krish Patel> if (ComparePythonVersion $versionString  $pythonMaxVersion) {
>>             return "correct version found"
>>         } else {
>> return "incorrect version found"
>> }
incorrect version found
PS C:\Users\Krish Patel> $versionString = "3.9.99"
PS C:\Users\Krish Patel> if (ComparePythonVersion $versionString  $pythonMaxVersion) {
>>             return "correct version found"
>>         } else {
>> return "incorrect version found"
>> }
incorrect version found
PS C:\Users\Krish Patel> $versionString = "3.10.0001"
PS C:\Users\Krish Patel> if (ComparePythonVersion $versionString  $pythonMaxVersion) {
>>             return "correct version found"
>>         } else {
>> return "incorrect version found"
>> }
correct version found
PS C:\Users\Krish Patel> $versionString = "3.10.12"
PS C:\Users\Krish Patel> if (ComparePythonVersion $versionString  $pythonMaxVersion) {
>>             return "correct version found"
>>         } else {
>> return "incorrect version found"
>> }
correct version found
```

**Comment by KrishPatel13** [comment]
Bug 1 and 2 (from https://github.com/OpenAdaptAI/OpenAdapt/issues/372#issue-1796969872):
image

Resolution Commit: [ed7f3e0](https://github.com/OpenAdaptAI/OpenAdapt/pull/409/commits/ed7f3e0e4df37a88bcd32caf14aeaaa8a6cf300b)

Resolution Transcript: 
https://github.com/OpenAdaptAI/OpenAdapt/issues/372#issuecomment-1642575497

File: install/install_openadapt.ps1 [hunk 1]
```diff
@@ -11,7 +11,8 @@ $tesseractInstallerLoc = "https://digi.bib.uni-mannheim.de/tesseract/tesseract-o
 $tesseractPath = "C:\Program Files\Tesseract-OCR"
 
 $pythonCmd = "python"
-$pythonVerStr = "Python 3.10*"
+$pythonMinVersion = "3.10.0" # Change this if a different Lower version are supported by OpenAdapt
```
Comment (KrishPatel13) [inline install/install_openadapt.ps1:14]: https://github.com/OpenAdaptAI/OpenAdapt/issues/372#issuecomment-1642658163

**Comment by KrishPatel13** [comment]
Bug 5 and 6 (from https://github.com/OpenAdaptAI/OpenAdapt/issues/372#issue-1796969872):
image
image


Resolution Commit: [cf0d21d](https://github.com/OpenAdaptAI/OpenAdapt/pull/409/commits/cf0d21dc3b20754fa5df489e28cf63b35c04e47b)

Resolution Documentation: 
https://github.com/microsoft/vscode-python/issues/2559#issuecomment-420746291

**Comment by KrishPatel13** [comment]
Bugfix 3 (update):

Commit: [56cf2c1](https://github.com/OpenAdaptAI/OpenAdapt/pull/409/commits/56cf2c17ae7f138f9b273b68a4e1ddbc65936779)

File: install/install_openadapt.ps1 [hunk 2]
```diff
@@ -266,6 +285,6 @@ RunAndCheck "poetry install" "Run ``poetry install``"
 RunAndCheck "poetry run alembic upgrade head" "Run ``alembic upgrade head``" -SkipCleanup:$true
 RunAndCheck "poetry run pytest" "Run ``Pytest``" -SkipCleanup:$true
 Write-Host "OpenAdapt installed Successfully!" -ForegroundColor Green
-Start-Process powershell -ArgumentList "-Command poetry shell"
+Start-Process powershell -Verb RunAs -ArgumentList "-NoExit", "-Command", "Set-Location -Path '$pwd'; poetry shell"
```
Comment (KrishPatel13) [inline install/install_openadapt.ps1:288]: image

File: install/install_openadapt.ps1 [hunk 3]
```diff
@@ -260,6 +260,8 @@ function GetGitCMD {
 
 ################################   SCRIPT    ################################
 
+Write-Host "Install Script Started..." -ForegroundColor Yellow
```
Comment (KrishPatel13) [inline install/install_openadapt.ps1:263]: I could not reproduce the bug #3, hence what we can do for now is to do add this line and next time if the script fails we can at least know that it had ran the first line (Write-Host simlar to echo) or not. 

We can put an issue for the same ii the script is exiting without any messages to ensure that this is still being an issue. :(

**Comment by KrishPatel13** [comment]
@abrichr Ready for review!

File: install/install_openadapt.ps1 [hunk 4]
```diff
@@ -11,7 +11,8 @@ $tesseractInstallerLoc = "https://digi.bib.uni-mannheim.de/tesseract/tesseract-o
 $tesseractPath = "C:\Program Files\Tesseract-OCR"
 
 $pythonCmd = "python"
-$pythonVerStr = "Python 3.10*"
+$pythonMinVersion = "3.10.0" # Change this if a different Lower version are supported by OpenAdapt
+$pythonMaxVersion = "3.10.12" # Change this if a different HIgher version are supported by OpenAdapt
```
Comment (abrichr) [inline install/install_openadapt.ps1:15]:
```suggestion
$pythonMaxVersion = "3.10.12" # Change this if a different Higher version are supported by OpenAdapt
```

**Comment by KrishPatel13** [comment]
@abrichr Ready for Final Review and merging! :)

**Pull request merged**